tests/e2e/framework/kubernetes.go

```diff
@@ -793,15 +795,29 @@ EOF`, k.KubeConfig, dummyValidationRule)
 	return nil
 }
 
-func (k *KubeInfo) deployIstioWithHelm() error {
-	yamlFileName := filepath.Join(istioInstallDir, helmInstallerName, "istio", "templates", "crds.yaml")
+func (k *KubeInfo) deployCRDs(kubernetesCRD string) error {
+	yamlFileName := filepath.Join(istioInstallDir, helmInstallerName, "istio-init", "files", kubernetesCRD)
 	yamlFileName = filepath.Join(k.ReleaseDir, yamlFileName)
 
 	// deploy CRDs first
 	if err := util.KubeApply("kube-system", yamlFileName, k.KubeConfig); err != nil {
 		log.Errorf("Failed to apply %s", yamlFileName)
 		return err
 	}
+	return nil
+}
+
+func (k *KubeInfo) deployIstioWithHelm() error {
+	// Note: When adding a CRD to the install, an entry is needed here as well
+	istioCRDFileNames := []string{"crd-10.yaml", "crd-11.yaml", "crd-certmanager-10.yaml"}
+
+	// deploy all CRDs in Istio first
+	for _, yamlFileName := range istioCRDFileNames {
+		if err := k.deployCRDs(yamlFileName); err != nil {
+			log.Errorf("Failed to apply all Istio-specific CRDs")
```

Comment:
should add the the error message, the actual error and the yaml fail causing the failure.

Reply:
In an error condition, line 804 are executed, printing out the yaml file name.  The failure reason is then returned up the stack in error via line 805 and 818.

Reply:
if you log error on line 817, providing actual value of error and the value of yamlFileName will help to understand what went wrong with this specific file. If you return just error, then you do not know which file out of a range in istioCRDFileNames caused the error condition.

Reply:
done.